homeassistant/components/hyperion/light.py

```diff
@@ -375,3 +615,7 @@ async def async_added_to_hass(self):
         # Load initial state.
         self._update_full_state()
         return True
+
+    async def async_will_remove_from_hass(self):
+        """Disconnect from server."""
+        await self._client.async_client_disconnect()
```

Comment:
Isn't the client shared across all entities ?

Reply:
No. Each entity has its own client. A Hyperion server supports multiple instances. The client connects to the server -- then selects an instance, and receives subscriptions, in the context of that instance. As an instance == an entity, there is thus one client per entity.